pkg/autodiscovery/autoconfig.go

```diff
@@ -332,6 +332,7 @@ func (ac *AutoConfig) pollConfigs() {
 			case <-ac.health.C:
 			case <-ac.configsPollTicker.C:
 				// check if services tags are up to date
+				ac.configResolver.m.Lock()
```

Comment:
Which of the function calls here is not thread-safe? Would it be possible to make that function call thread-safe instead? It is not clear why the mutex is necessary here since `processDelService` and `processNewService` previously were thread-safe.

What about using another mutex that belongs to `AutoConfig`?

Reply:
@devonboyer because of the line below `for _, service := range ac.configResolver.services` we're both reading and writing to the map through the `processDelService`/`processNewService` functions.
That's also why I need a unsafe version otherwise the lock would conflict.

Reply:
Hopefully we'll move soon the service listening logic away from the `configresolver` so that will clean that up